browser/ui/views/infobars/brave_global_infobar_manager.cc

```diff
@@ -0,0 +1,192 @@
+      }
+    }
+  }
+}
+}  // namespace
+
+BraveGlobalInfoBarManager::BraveGlobalInfoBarManager(
+    std::unique_ptr<BraveGlobalConfirmInfoBarDelegateFactory> delegate_factory)
+    : delegate_factory_(std::move(delegate_factory)) {}
+
+BraveGlobalInfoBarManager::~BraveGlobalInfoBarManager() {
+  RemoveObserverForInfobarManagers(delegate_factory_->GetInfoBarIdentifier(),
+                                   this);
+}
+
+void BraveGlobalInfoBarManager::Show() {
+  DCHECK(is_closed_);
+  if (!is_closed_) {
+    return;
+  }
+  is_closed_ = false;
+  browser_tab_strip_tracker_ =
+      std::make_unique<BrowserTabStripTracker>(this, nullptr);
+  browser_tab_strip_tracker_->Init();
+}
+
+void BraveGlobalInfoBarManager::TabChangedAt(content::WebContents* contents,
+                                             int index,
+                                             TabChangeType change_type) {
+  infobars::ContentInfoBarManager* infobar_manager =
+      infobars::ContentInfoBarManager::FromWebContents(contents);
+  MaybeAddInfoBar(infobar_manager);
+}
+
+void BraveGlobalInfoBarManager::OnTabStripModelChanged(
+    TabStripModel* tab_strip_model,
+    const TabStripModelChange& change,
+    const TabStripSelectionChange& selection) {
+  if (!selection.new_contents || is_closed_) {
+    return;
+  }
+
+  infobars::ContentInfoBarManager* new_infobar_manager =
+      infobars::ContentInfoBarManager::FromWebContents(selection.new_contents);
+  if (!new_infobar_manager) {
+    return;
+  }
+
+  MaybeAddInfoBar(new_infobar_manager);
+
+  if (!selection.old_contents || !selection.active_tab_changed()) {
+    return;
+  }
+
+  infobars::ContentInfoBarManager* old_infobar_manager =
+      infobars::ContentInfoBarManager::FromWebContents(selection.old_contents);
+
+  if (!old_infobar_manager) {
+    return;
+  }
+  old_infobar_manager->RemoveObserver(this);
+  RemoveInfobarsByIdentifier(old_infobar_manager,
+                             delegate_factory_->GetInfoBarIdentifier(), this);
+}
+
+void BraveGlobalInfoBarManager::MaybeAddInfoBar(
+    infobars::InfoBarManager* infobar_manager) {
+  DCHECK(infobar_manager);
+  if (!infobar_manager) {
+    return;
+  }
+  auto delegate = delegate_factory_->Create();
+  if (!delegate) {
+    is_closed_ = true;
+    return;
+  }
+
+  delegate->AddObserver(this);
+
+  if (infobars::InfoBar const* added_bar = infobar_manager->AddInfoBar(
+          CreateBraveGlobalInfoBar(std::move(delegate)));
+      !added_bar) {
+    return;
+  }
+
+  infobar_manager->RemoveObserver(this);
+  infobar_manager->AddObserver(this);
+}
+
+void BraveGlobalInfoBarManager::OnInfoBarClosed() {
+  infobars::InfoBarManager* current_infobar_manager =
+      GetCurrentInfoBarManager();
+  if (!current_infobar_manager) {
+    return;
+  }
+  OnManagerShuttingDown(current_infobar_manager);
+}
+
+void BraveGlobalInfoBarManager::OnManagerShuttingDown(
+    infobars::InfoBarManager* manager) {
```

Comment:
Why do we need to pass manager here? Just remove all observers via RemoveObserverForInfobarManagers and you can remove unpredictable Browser::GetLastActive() call